src/DiffSharp.Core/Model.fs

```diff
@@ -80,15 +80,28 @@ type ParameterDict() =
     /// <summary>TBD</summary>
     member d.iter(f:string*Parameter->unit) = for KeyValue(n, p) in d.values do f(n,p)
 
-    /// <summary>TBD</summary>
+    /// <summary>
+    ///  Adjust the parameters to include support for forward-mode automatic differentiation.
+    /// </summary>
+    /// <param name="derivatives">The derivatives of the parameters</param>
+    /// <param name="tag">The level tag for nested differentiation.  Defaults to the current global nesting level</param>
+    /// <remarks>
+    ///  After this call the current parameters of the model will have attached derivatives for forward mode propagation.
+    /// </remarks>
     member d.forwardDiff(derivatives:ParameterDict, ?tag:uint32) = 
-        let tag = defaultArg tag GlobalNestingLevel.Current
-        d.iter(fun (n, p) -> p.forwardDiff(derivatives.[n], tag))
-
-    /// <summary>TBD</summary>
+        d.iter(fun (n, p) -> p.forwardDiff(derivatives.[n], ?tag=tag))
+
+    /// <summary>
+    ///  Adjust the parameters to include support for reverse-mode automatic differentiation.
+    /// </summary>
+    /// <param name="tag">The level tag for nested differentiation.  Defaults to the current global nesting level</param>
+    /// <remarks>
+    ///  After this call the current parameters of the model will support reverse-mode propagation. After the completion
+    ///  of the corresponding <c>reverse</c> operation, the computed derivative
+    ///  will be available. 
+    /// </remarks>
     member d.reverseDiff(?tag:uint32) = 
-        let tag = defaultArg tag GlobalNestingLevel.Current
```

Comment:
Note the default is applied in p.reverseDiff

Reply:
Please see the comment I left for the `forwardDiff` case.